src/protocols/XDGShell.cpp

```diff
@@ -534,30 +534,29 @@ CBox CXDGPositionerRules::getPosition(const CBox& constraint, const Vector2D& pa
     if (success)
         return predictedBox.translate(-parentCoord - constraint.pos());
 
+    CBox test = predictedBox;
+
     if (state.constraintAdjustment & (XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X | XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y)) {
         // attempt to flip
         const bool flipX = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X;
         const bool flipY = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y;
 
-        CBox       test = predictedBox;
-        success         = true;
-        if (flipX && test.copy().translate(Vector2D{-predictedBox.w - state.anchorRect.w, 0}).expand(-1).inside(constraint))
+        const bool horizontalEdgeOut = test.x < constraint.x || test.x + test.w > constraint.x + constraint.w;
+        const bool verticalEdgeOut   = test.y < constraint.y || test.y + test.h > constraint.y + constraint.h;
+
+        if (flipX && horizontalEdgeOut && flipY && verticalEdgeOut)
```

Comment:
this still doesn't test if flipping fixes this, confront this professional drawing:
![image](https://github.com/hyprwm/Hyprland/assets/43317083/d727cbfc-5de6-4070-9ff6-b2757d592771)

Reply:
But the flip would just invert the x direction. I just tested this to be sure.
In the picture it would mean the popup gets repositioned to the other side, which is not the case.